src/main/java/org/peergos/client/NabuClient.java

```diff
@@ -77,20 +77,30 @@ public Map id() throws IOException {
         return retrieveMap("id");
     }
 
-    public List<Multihash> localRefs() throws IOException {
+    public List<Multihash> getRefs() throws IOException {
         String jsonStream = new String(retrieve("refs/local"));
         return JSONParser.parseStream(jsonStream).stream()
                 .map(m -> (String) (((Map) m).get("Ref")))
                 .map(Cid::decode)
                 .collect(Collectors.toList());
     }
 
-    public byte[] getBlock(Multihash hash) throws IOException {
-        return retrieve("block/get?stream-channels=true&arg=" + hash);
+    public boolean hasBlock(Multihash hash, Optional<String> auth) throws IOException {
+        String authArg = auth.isPresent() ? "&auth=" + auth.get() : "";
+        return "true".equals(new String(retrieve("block/has?arg=" + hash + authArg)));
+    }
+
+    public boolean bloomAdd(Multihash hash) throws IOException {
+        return "true".equals(new String(retrieve("bloom/add?arg=" + hash)));
+    }
+
+    public byte[] getBlock(Multihash hash, Optional<String> auth) throws IOException {
+        String authArg = auth.isPresent() ? "&auth=" + auth.get() : "";
+        return retrieve("block/get?arg=" + hash + authArg);
     }
 
     public byte[] removeBlock(Multihash hash) throws IOException {
```

Comment:
Is this returning the block? I don't think we need that. It can be void.

Reply:
it returns a json structure with error and cid. not required. have made return type null.